src/Compilers/.editorconfig

```diff
@@ -1,6 +1,6 @@
 # Project-specific settings:
 # CSharp code style settings:
 [*.cs]
-csharp_style_var_for_built_in_types = false:none
+csharp_style_var_for_built_in_types = false:warning
 csharp_style_var_when_type_is_apparent = true:none
-csharp_style_var_elsewhere = false:none
\ No newline at end of file
+csharp_style_var_elsewhere = false:warning
```

Comment:
This says warning. Convivce me it's not a warning. If it isn't a warning why is ti called warning?

Reply:
> Convivce me it's not a warning.

The CI build passed and I did not make any changes to the code in this directory. If my previous statement was not correct, the build would have failed.

> If it isn't a warning why is ti called warning?

The setting controls the color of the squiggle used in the editor when working on an open file. This setting is non-blocking for a user's workflow but gently alerts them to *problems* prior to them reaching your team members for review.

Reply:
I meant that it won't be a warning in the IDE. I don't want to take a change right now taht will have a big negative impact on the IDE experience for our developers. We've chatted about this type of change in broader meetings (not copmiler team) and there was push back in doing this now.

Reply:
@jaredpar Could you clarify that a bit more?  How would this be worse for the developer experience vs the current situation where the PR is submitted, but then *that's* the point where the issue is noticed, reported on, and the dev has to go fix it up anyways?

Wouldn't it be better to get the info early, so they can know and fix, prior to the PR?

Reply:
> Wouldn't it be better to get the info early, so they can know and fix, prior to the PR?

This would add tens of thousands of alerts to a compiler developer on our current code base. Yes it depends on files open but it's still going to be a pretty noisy change. We're not taking the equivalent of a "fix all" at the moment for a couple of reasons. One of which being the style enforced here doesn't actually reflect the style we enforce. Differs in a number of areas like `foreach`. 

I definitely think the negative experience here outweighs the small number of times `var` is asked to be undone during a review. In particular because it will have a much stronger impact on the compiler team (who aren't pushing for this change) than the people who are pushing for it. 

I do want us to embrace editorconfig here. But at this point the style we tend to enforce doesn't line up with any of the listed styles. We need to look at this as a part of the up coming editorconfig work and decide what we want to do as a team.

Reply:
>  One of which being the style enforced here doesn't actually reflect the style we enforce. Differs in a number of areas like foreach.

Can you expand on this?  It would be good to know what the actual style that's preferred is so that we can attempt to support that sort of thing in the IDE featureset.

Reply:
If you can point at some examples, and explain why you don't lke them, i'd be happy to file issues on the IDE to examine what we could do to better make this feature area fit other projects that might like the same style as you.